### DISCUSSION TITLE
0.11.0

### DISCUSSION THREAD

**Discussion by MohamedBassem** [karakeep-app/karakeep#73] [Announcements]
Welcome to the v0.11.0 release of Hoarder. This packs a bunch of new stuff and fixes.

## New Features
- There's now a CLI tool to enable mass manipulations for your bookmarks. You can use it, for example, to mass import bookmarks from your previous tool. Check its docs [here](https://docs.hoarder.app/command-line), and you can check the initial version of importing bookmarks docs [here](https://docs.hoarder.app/import).
- You can now attach notes to your bookmarks.
- You can now delete tags in the tag view page.
- The `All Tags` page now differentiates between manually attached tags and AI attached tags.
- You can specify the language of the auto generated tags with `INFERENCE_LANG=<lang>` env variable. It currently defaults to `english`. Thanks @COOHU-Kr for the suggestion!
- The web UI now allows uploading multiple images at the same time. `Webp` support also got added.
- Clicking `⌘ + E` or (`ctrl + E`) will now focus on the new bookmark editor. Thanks @chanunnaki for the idea!
- Current server version is now shown in the admin panel. It'll also check github API to know if there are new version. You can disable the new version check with `DISABLE_NEW_RELEASE_CHECK=true`.
- In case you missed it, the android app is almost there (under google's mercy). If you can't wait, you can download the APK from [this issue](https://github.com/MohamedBassem/hoarder-app/issues/6)

## Fixes
- Search results are now sorted by relevance.
- In some rare instances, the crawler was getting stuck parsing some pages preventing any other links from getting crawled. There's now a 30s timeout to solve this problem.
- Fix the `sharp is missing` error in the `web` container logs.
- When logging out, people were getting redirected to `localhost:3000` by mistake. Add `NEXTAUTH_URL=<server_url>` to your env file to fix that. The installation docs has been updated.

## Upgrade
- [Optional but hightly recommended] The initial docker compose was using an old chrome version (`v100`). Bump the chrome container version to `v123` by changing the chrome container image:
  - from: `gcr.io/zenika-hub/alpine-chrome:100`
  - to: `gcr.io/zenika-hub/alpine-chrome:123`
- If you're using `HOARDER_VERSION=release`, run `docker compose pull && docker compose up -d`.
- If you're pinning it to a specific version, upgrade the version and then run `docker compose pull && docker compose up -d`.

### The new `All Tags` page

<img width="1292" alt="Screenshot 2024-04-02 at 5 20 16 PM" src="https://github.com/MohamedBassem/hoarder-app/assets/2418637/b99850e0-25e6-4b99-aaf6-f9ae0977731f">

### Attaching notes to bookmarks
<img width="363" alt="Screenshot 2024-04-02 at 5 20 49 PM" src="https://github.com/MohamedBassem/hoarder-app/assets/2418637/f6ac2a64-2c16-45d4-9987-8e536c35e438">

### Server version in admin panel
<img width="514" alt="Screenshot 2024-04-02 at 5 20 28 PM" src="https://github.com/MohamedBassem/hoarder-app/assets/2418637/7605f9c2-3196-4d52-a081-9ef72a541b0d">

### The new CLI
<img width="758" alt="Screenshot 2024-04-02 at 5 24 25 PM" src="https://github.com/MohamedBassem/hoarder-app/assets/2418637/0106253f-276d-47d4-8b5c-58061fd56d16">

<hr /><em>This discussion was created from the release <a href='https://github.com/MohamedBassem/hoarder-app/releases/tag/v0.11.0'>0.11.0</a>.</em>

**Comment by victornavorskie** [comment]
is there are any way to import bookmarks from browsers ?

**Reply by MohamedBassem** [reply]
I documented a non-trivial way to import bookmarks from chrome: https://docs.hoarder.app/import. It'll require some technical knowledge though, but I'm happy to help.

**Comment by ovizii** [comment]
> You can now delete tags in the tag view page.

Can you elaborate if there's any learning going on here? Is there any correlation between my own tags and the AI tags? What happens if I delete an AI tag? Will it reconsider when to use that specific tag? Will it stop using it? Will re-index or re-crawl trigger tagging and will it simply result in the same (deleted) tag again?

I hope I didn't come across as rude throwing all these questions out there, I'm simply curious how it all works :-)

**Reply by MohamedBassem** [reply]
haha, no worries :) Deleting a tag is simply equivalent to removing that tag from all the bookmarks that currently have it. No change in behavior between manual tags and AI tags. There's also no blocklisting happening here. A future bookmark (or a re-crawl) might result in this tag appearing again.

We can probably build tag-blocklisting mechanisms to prevent AI from adding specific tags, but to be honest, I don't yet see the usecase for that. If there's a tag I don't care about, I'd probably just ignore it. What do you think?

**Reply by ovizii** [reply]
Thanks for the reply. Let me add my thoughts to your question:

> If there's a tag I don't care about, I'd probably just ignore it. What do you think?

I think that really depends on how you use tags. Maybe you could share that?

I usually see tags used in conjunction with categories (in the case of hoarder that seems to be the lists you have implemented). 
The apps I used to use, do have some filtering implemented where I go search for a tag and filter by category or search and then filter by category and tag.

Say I search for stir-fry and then filter by category: recipes and tag: pork.

Or I filter by tag: server and limit to category: technology 

Currently, the AI seems to massively tag every article. How about an env variable to limit the number of tags per bookmark? 
Also, feel free to split this discussion out of this release announcement if you like, I really should have opened a separate thread.

**Reply by MohamedBassem** [reply]
For me, tags was a way to augment the piece of data that I saved with some of the stuff that are not referenced literally in the original content for the purpose of search. For example with only that one line:

<img width="352" alt="Screenshot 2024-04-03 at 2 43 56 PM" src="https://github.com/MohamedBassem/hoarder-app/assets/2418637/085cc288-7b54-4385-83f8-b42247c5b8ac">

Hoarder figured out that this is a recipe, and that it's for pancakes. Without the automatic tags, searching for "recipe" wouldn't have yielded that result. That's why in [the prompt](https://github.com/MohamedBassem/hoarder-app/blob/main/apps/workers/openaiWorker.ts#L81), I wasn't too strict on the quality of the tags. My thoughts was that the more the better. And for images, I actually let hoarder produce a ton of tags because that's the only way you can reference that image in search.

Looking at your usecase, it seems that having an excessive number of tags also won't hurt your filtering capabilities right?

I think this discussion is useful and indeed worth it's own thread for visibility. Let's indeed open a separate thread for it.

**Reply by ovizii** [reply]
I see, thanks for sharing how you use tags. Theoretically, too many tags won't hurt but if the AI makes a tag out of half of the text, that makes searching pointless? I saw tags and lists (categories) as a means of further narrowing down a search.

Anyway, this brings me back to searching. Do you plan on adding more advanced options to the search? I also like the option to save useful searches (readeck calls them collections). In this example, as my workflow with readeck differed, I saved these collections which filters all un-tagged bookmarks. I used to wildly bookmark, then read them later and manually tag them. 

With hoarder, due to AI tagging this seems pointless, but I am still looking for a way to achieve this, so have you given any thoughts to mark bookmarks as read/unread? At least in the context of read-later this makes sense although less so in the context of bookmarks.

i.e. 

![image](https://github.com/MohamedBassem/hoarder-app/assets/837902/7fdaf6f8-85d6-4a49-b1b2-a4642fd03c10)

**Reply by MohamedBassem** [reply]
> Do you plan on adding more advanced options to the search?

Advanced search (maybe not as advanced as your screenshot) is planned. You'll be able to filter with tags and lists and decide whether or not to include archived (and maybe favourited as well because why not). I don't currently have an issue tracking it though, please don't hesitate to open one (if you don't, I will).
 
> I also like the option to save useful searches

This is being tracked in https://github.com/MohamedBassem/hoarder-app/issues/65. Although the title says based on `tags`, I'm planning to make it saved searches in general.

> At least in the context of read-later this makes sense although less so in the context of bookmarks.

Actually, marking something as read/done is useful even in the bookmarks usecase. I sometimes bookmark stuff to check them out later, and I want to mark them as `checked` once I'm done with them (but I want to still keep them in my library). And I 100% agree with you that there's no neat way to do it right now in hoarder and this is something that I personally need and planning to add as soon as in the next release. I currently have a couple of ideas:

Option 1: Introduce a new `Remove from homepage` option for a bookmark. That way the homepage becomes my "inbox" and contains everything there requires my attention. Once I'm done with an item, I drop it in a list and remove it from my homepage.

Option 2: Explicitly introduce read/unread flags on bookmarks. Everything is `unread` until explicitly marked as `read`.

Option 3: Introduce a configurable default list where all bookmarks get added to, and then you remove the items that no longer require attention from that list once you're done with them.

Not 100% satisfied with any of them yet, so I'm open to ideas :)

**Reply by ovizii** [reply]
All 3 options sound workable to me. No preference or further suggestions from my side. I like the “roadmap” I can infer from our few interactions we had. :-)
